packages/block-editor/src/hooks/use-bindings-attributes.js

```diff
@@ -73,18 +75,28 @@ const createEditFunctionWithBindingsAttribute = () =>
 
 							if ( metaValue ) {
 								updatedAttributes[ attributeName ] = metaValue;
+								hasSource = true;
 							}
 						}
 					}
 				);
 			}
 
 			return (
-				<BlockEdit
-					key="edit"
-					{ ...props }
-					attributes={ updatedAttributes }
-				/>
+				// TODO: only set this if the block is connected to a source.
+				// this might not be a good way to do it if a block can have child
+				// blocks that are NOT connected to a source.
+				<div
+					style={
+						hasSource ? { '--wp-admin-theme-color': '#9747FF' } : {}
+					}
+				>
```

Comment:
I'm not quite sure if this is the best approach to add color to the focus outline.

Reply:
This extra containing `div` causes the in-between block inserter to break, am currently seeing how else this could be done.

Reply:
Having a wrapping `div` is a blocker here because it will impact existing CSS styles. Additional styles or having it abstracted as a class name can be done inside `useBlockProps`.

By the way, failing e2e tests reported on CI might be also caused by this change or by the fact that there is now an additional item in the block toolbar proposed.

Reply:
@gziolo I've moved the styles to `useBlockProps` in https://github.com/WordPress/gutenberg/pull/59185/commits/75f19c37ec557585e04452cc2b774a842841d4e1.

How does this look?